client/src/ProvGraph.tsx

```diff
@@ -0,0 +1,102 @@
+import { Component } from 'react';
+import Location from './Models/Location';
+import {Dropdown, DropdownButton} from 'react-bootstrap';
+import { LineChart, Line, XAxis, YAxis, Label, Tooltip } from 'recharts';
+import { extent as d3Extent, max as d3Max } from 'd3-array';
+import { scaleLinear as d3ScaleLinear, scaleTime as d3ScaleTime, tickFormat} from 'd3-scale';
+import { format as d3Format } from 'd3-format';
+
+export default class ProvGraph extends Component<{longitude: any, latitude: any}, {prov: any, location: Location, selectedData: string}> {
+    async componentWillReceiveProps(nextProps: {latitude: any, longitude: any}): Promise<void> {
+        if (nextProps.latitude !== 0 && nextProps.longitude !== 0) {
+            this.setState({prov: await this.getCovidStatsByLocation(nextProps.latitude, nextProps.longitude)});
+            this.setState({location: await this.getLocationFromPosition(nextProps.latitude, nextProps.longitude)});
+        }
+    }
+
+    componentWillUnmount() {
+        this.setState = (state,callback)=>{
+            return;
+        };
+    }
+    
+    private getTime(date?: Date) {
+        return date != null ? date.getTime() : 0;
+    }
+
+    async getCovidStatsByLocation(latitude: number, longitude: number): Promise<any> {
+
+        try {
+            const url = `http://localhost:8080/location?longitude=${longitude.toPrecision(5)}&latitude=${latitude.toPrecision(5)}&component=provgraph`;
+            let result: any = await (await fetch(url)).json();//.json();
+            return result;
+        }
+        catch (e) {
+            console.log(e);
+        }
+    }
+    
+    async getLocationFromPosition(latitude: number, longitude: number): Promise<any> {
+
+        try {
+            const url = `http://localhost:8080/location?longitude=${longitude.toPrecision(5)}&latitude=${latitude.toPrecision(5)}&component=locationdata`;
+            let result: any = await (await fetch(url)).json();//.json();
+            return result;
+        }
+        catch (e) {
+            console.log(e);
+        }
+    }
+
+    render() {
+        let destinationObj = [{}];
```

Comment:
Extract mapping to function. Do not do logic in the render.

Reply:
Can I please leave the weird d3 stuff that does the graph tickers and formatting in render I have no idea how to extract that

Reply:
All logic has been moved to private functions, but React yells at you if you try to call the functions in the JSX directly so I just assigned them to variables in the render method. Hope that's ok, if not I can reassign to global variables or something, that just seemed like more of a pain in the ass.